include/Dialect/Openfhe/IR/OpenfheTypes.td

```diff
@@ -0,0 +1,55 @@
+#ifndef INCLUDE_DIALECT_OPENFHE_IR_OPENFHETYPES_TD_
+#define INCLUDE_DIALECT_OPENFHE_IR_OPENFHETYPES_TD_
+
+include "OpenfheDialect.td"
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+
+class Openfhe_Type<string name, string typeMnemonic, list<Trait> traits = []>
+    : TypeDef<Openfhe_Dialect, name, traits> {
+  let mnemonic = typeMnemonic;
+}
+
+// Plaintext
+class Openfhe_Plaintext<string scheme>
+  : Openfhe_Type< "Plaintext" # scheme, "plaintext" # scheme> {
+  let summary = "A plaintext corresponding to OpenFHE's ciphertext type in " # scheme # " scheme";
+}
+
+foreach scheme = ["BFV", "BGV", "CKKS"] in {
+  def Openfhe_Plaintext # scheme : Openfhe_Plaintext<scheme>;
+}
+
+def Openfhe_PlaintextType :
+  AnyTypeOf<[
+    Openfhe_PlaintextBFV,
+    Openfhe_PlaintextBGV,
+    Openfhe_PlaintextCKKS,
+  ]>;
+
+// Ciphertext
+class Openfhe_Ciphertext<string scheme>
+  : Openfhe_Type< "Ciphertext" # scheme, "ciphertext" # scheme> {
+  let summary = "A ciphertext corresponding to OpenFHE's ciphertext type in " # scheme # " scheme";
+}
+
+foreach scheme = ["BFV", "BGV", "CKKS"] in {
+  def Openfhe_Ciphertext # scheme : Openfhe_Ciphertext<scheme>;
+}
+
+def Openfhe_CiphertextType :
+  AnyTypeOf<[
+    Openfhe_CiphertextBFV,
+    Openfhe_CiphertextBGV,
+    Openfhe_CiphertextCKKS,
+  ]>;
+
+def Openfhe_ServerKey : Openfhe_Type<"ServerKey", "server_key"> {
```

Comment:
In the OpenFHE API they seem to call this type `CryptoContext`. Would you mind re-naming it to `Openfhe_CryptoContext`? I think it would help the user of this dialect understand the intended purpose of this type. In a perfect world, we can make this exit dialect a 1-1 mirror of the OpenFHE API.

Reply:
Do you mean changing `Openfhe_ServerKey` to `Openfhe_CryptoContext`?

Reply:
Exactly, just a name change to make it closer to the OpenFHE API.

Reply:
Applied!